engine/api/workflow_queue.go

```diff
@@ -217,6 +217,34 @@ func takeJob(ctx context.Context, dbFunc func() *gorp.DbMap, store cache.Store,
 		return nil, sdk.WithStack(err)
 	}
 
+	secretsReqs := job.Job.Action.Requirements.FilterByType(sdk.SecretRequirement).Values()
+	secretsReqsRegs := make([]*regexp.Regexp, 0, len(secretsReqs))
+	for i := range secretsReqs {
+		r, err := regexp.Compile(secretsReqs[i])
+		if err != nil {
+			return nil, sdk.WithStack(err)
+		}
+		secretsReqsRegs = append(secretsReqsRegs, r)
+	}
+
+	// Filter project's secrets depending of the region requirement that was set on job
```

Comment:
it could be nice to add a log info to track when it happen

Reply:
Added spawn info.